File: backend/src/main/java/org/kaiteki/backend/tasks/models/dto/UpdateTaskDTO.java
package org.kaiteki.backend.tasks.models.dto;

import lombok.Builder;
import lombok.Data;
import org.kaiteki.backend.tasks.models.TaskPriority;

import java.time.LocalDate;

@Data
@Builder
public class UpdateTaskDTO {
    private String title;
    private String content;
    private String description;
    private String tag;
    private LocalDate endDate;
    private LocalDate startDate;
    private TaskPriority priority;
    private Long statusId;
    private Long executorId;
}

File: frontend/src/app/teams/submodules/tasks/models/update-task.dto.ts
import { TaskPriority } from './tasks.model';

export interface UpdateTaskDTO {
  title?: string;
  content?: string;
  description?: string;
  tag?: string;
  endDate?: Date;
  startDate?: Date;
  priority?: TaskPriority;
  statusId?: number;
  executorId?: number;
}
